Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt

```diff
@@ -41,23 +46,54 @@ object MainDestinations {
 }
 
 @Composable
-fun NavGraph(startDestination: String = MainDestinations.ONBOARDING_ROUTE) {
-    val navController = rememberNavController()
+fun NavGraph(
+    modifier: Modifier = Modifier,
+    finishActivity: () -> Unit = {},
+    navController: NavHostController = rememberNavController(),
+    startDestination: String = MainDestinations.COURSES_ROUTE,
+    showOnboardingInitially: Boolean = true
+) {
+    // Onboarding could be read from shared preferences.
+    val onboardingComplete = remember(showOnboardingInitially) {
+        mutableStateOf(!showOnboardingInitially)
+    }
 
     val actions = remember(navController) { MainActions(navController) }
+
     NavHost(
         navController = navController,
         startDestination = startDestination
     ) {
         composable(MainDestinations.ONBOARDING_ROUTE) {
-            Onboarding(onboardingComplete = actions.onboardingComplete)
+            // Intercept back in Onboarding: make it finish the activity
+            BackHandler {
+                finishActivity()
+            }
+
+            Onboarding(
+                onboardingComplete = {
```

Comment:
Trailing lambda?

Reply:
When the function of the trailing lambda is not obvious like in this case I prefer to keep the name